spack.yaml

```diff
@@ -5,26 +5,36 @@
 spack:
   # add package specs to the `specs` list
   specs:
-    - access-om2@git.2024.03.0
+    - access-om2@git.2024.03.1
   packages:
     cice5:
-      require: '@git.2023.10.19'
+      require:
+        - '@git.2023.10.19'
     mom5:
-      require: '@git.2023.11.09'
+      require:
+        - '@git.2023.11.09'
     libaccessom2:
-      require: '@git.2023.10.26'
+      require:
+        - '@git.2023.10.26'
     oasis3-mct:
-      require: '@git.2023.11.09'
+      require:
+        - '@git.2023.11.09'
     netcdf-c:
-      require: '@4.7.4'
+      require:
+        - '@4.9.2'
     netcdf-fortran:
-      require: '@4.5.2'
+      require:
+        - '@4.6.1'
     parallelio:
-      require: '@2.5.2'
+      require:
+        - '@2.6.2'
     openmpi:
-      require: '@4.0.2'
+      require:
+        - '@4.1.5'
+        - 'cppflags="-diag-disable=10441"'
     all:
-      compiler: [intel@19.0.5.281]
+      compiler: [intel@2021.10.0]
+      target: [x86_64]
```

Comment:
Hi @CodeGat, In the dev container, Spack used `intel@2021.2.0` instead of `2021.10.0`. I suspect this is because the syntax above indicates a preference, not a requirement. e.g. `packages.yaml`
```
  cmake:
    # Syntax for requirement:
    require: "@3.24.4"
    # Syntax for preference:
    #version: [3.24.4]
```

Reply:
The compilers available in the dev container I was using:
```
# spack compiler list
==> Available compilers
-- dpcpp rocky8-x86_64 ------------------------------------------
dpcpp@2023.2.0  dpcpp@2022.1.0  dpcpp@2021.2.0

-- gcc rocky8-x86_64 --------------------------------------------
gcc@8.5.0

-- intel rocky8-x86_64 ------------------------------------------
intel@2021.6.0  intel@2021.2.0  intel@2021.10.0

-- oneapi rocky8-x86_64 -----------------------------------------
oneapi@2023.2.0  oneapi@2022.1.0  oneapi@2021.2.0
```

Reply:
Good point...what a pain...Will batch this in with the restricted versions of `netcdf`/etc @penguian noted earlier

Reply:
Updated to use `require:` rather than `compiler:`/`target:` in https://github.com/ACCESS-NRI/ACCESS-OM2/pull/60/commits/c49a105908d07edd074edf421a289588e2977f2a

Reply:
That worked, `intel@2021.10.0` was used in the dev container.